sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/UnsupportedOperationChecker.scala

```diff
@@ -41,23 +42,49 @@ object UnsupportedOperationChecker extends Logging {
     }
   }
 
+  def hasRangeExpr(e: Expression): Boolean = e.exists {
```

Comment:
The method name is too general than what it actually does. Maybe `hasRangeExprAgainstEventTimeCol` ?
Or just define the method from the start of `isStreamStreamIntervalJoin` so that it is under the proper implicit context.

Reply:
Done, put it under isStreamStreamIntervalJoin. Also add `private` before the functions to limit the scope.